hr_attendance_reason/controllers/main.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2024 Tecnativa - Carlos Lopez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import http
+from odoo.http import request, route
+
+from odoo.addons.hr_attendance.controllers.main import HrAttendance
+
+
+class HrAttendance(HrAttendance):
+    # inherited routes
+    @route("/hr_attendance/attendance_user_data", type="json", auth="user")
+    def user_attendance_data(self):
+        response = super().user_attendance_data()
+        # try to get the company of the employee to show the correct reasons from
+        # the employees company
+        company = (
+            request.env.user.employee_id.company_id
+            if request.env.user.employee_id
+            else request.env.company
+        )
+        response.update(self._get_attendance_reason_settings(company))
+        # get available reasons for employee company
+        reasons = []
+        if response.get("attendance_state", False):
+            if response.get("attendance_state", False) == "checked_out":
+                action_type = "sign_in"
+            else:
+                action_type = "sign_out"
+            reasons = self._get_attendance_reasons(action_type, company)
+        response.update({"reasons": reasons})
+        return response
+
+    @route("/hr_attendance/systray_check_in_out", type="json", auth="user")
+    def systray_attendance(self, latitude=False, longitude=False):
+        if request.params.get("attendance_reason_id"):
+            request.update_context(
+                attendance_reason_id=request.params.get("attendance_reason_id")
+            )
+        return super().systray_attendance(latitude=latitude, longitude=longitude)
+
+    @http.route("/hr_attendance/manual_selection", type="json", auth="public")
+    def manual_selection_with_geolocation(
+        self, token, employee_id, pin_code, latitude=False, longitude=False
```

Comment:
Why do you need `latitude` and `longitude` fields in the migration? That change is not necessary, if you need it, you can use `hr_attendance_geolocation` module.

In addition, the method name must be left as before `manual_selection`

Reply:
No I do not need this fields but in Odoo 18 the method "manual_selection_with_geolocation" has now the route "/hr_attendance/manual_selection" (in Odoo 17 the route was on the method "manual_selection"). So in my option I need to overwrite this method to ensure the same functionality compare to Odoo 17.

Reply:
Sorry but I can not find any reference for "manual_selection_with_geolocation" metodo on odoo 18 , can you help me?